Dịch "chapter_appendix_math/single-variable-calculus_vn" - Phần 03 (#435)
aivivn/d2l-vn: chapter_appendix_math/single-variable-calculus_vn.md

For instance, suppose that we know that $L(x) = x^{2} + 1701(x-4)^3$, then we can see how large this value is at the point $x = 4$ as follows.
-->

> Những kiến thức trên đã là đủ để chúng ta bắt đầu thực hành viết mã.

Reviewer: ```suggestion
Với những kiến thức trên đã đủ để chúng ta bắt đầu thực hành viết mã.
```

> Những kiến thức trên đã là đủ để chúng ta bắt đầu thực hành với mã lập trình.

Reviewer 2: ```suggestion
Những kiến thức trên đã là đủ để chúng ta bắt đầu thực hành viết mã.
```

> Ví dụ, giả sử ta có $L(x) = x^{2} + 1701(x-4)^3$, thì chúng ta có thể biết được độ lớn của giá trị này tại điểm $x = 4$ như sau:

Reviewer: ```suggestion
Ví dụ, giả sử ta có $L(x) = x^{2} + 1701(x-4)^3$, thì ta có thể biết được độ lớn của giá trị này tại điểm $x = 4$ như sau:
```

> Ví dụ, giả sử ta có $L(x) = x^{2} + 1701(x-4)^3, thì chúng ta có thể biết được độ lớn của giá trị này tại điểm $x = 4$ như sau:

Reviewer 2: ```suggestion
Ví dụ, giả sử ta có $L(x) = x^{2} + 1701(x-4)^3$, thì chúng ta có thể biết được độ lớn của giá trị này tại điểm $x = 4$ như sau:
```

<!--
Now, if we are observant, we will notice that the output of this number is suspiciously close to $8$.  
Indeed, if we decrease $\epsilon$, we will see value becomes progressively closer to $8$.  

> Thus we may conclude, correctly, that the value we seek (the degree a change in the input changes the output) should be $8$ tại điểm $x=4$..  

Reviewer 2: Chỗ này bị gì vậy ông giáo?

Author: > Chỗ này bị gì vậy ông giáo?

Chỗ này bị ngáo ông giáo ạ :))

> Thus we may conclude, correctly, that the value we seek (the degree a change in the input changes the output) should be $8$ tại điểm $x=4$..  

Author: ```suggestion
Thus we may conclude, correctly, that the value we seek (the degree a change in the input changes the output) should be $8$ at the point $x=4$.  
```

The way that a mathematician encodes this fact is
-->

> Bây giờ bằng việc quan sát, chúng ta sẽ nhận ra rằng kết quả của con số này là xấp xỉ $8$.

Reviewer 2: ```suggestion
Nếu để ý kĩ, chúng ta sẽ nhận ra rằng kết quả của con số này là xấp xỉ $8$.
```

The way that a mathematician encodes this fact is
-->

Nếu để ý kĩ, chúng ta sẽ nhận ra rằng kết quả của con số này là xấp xỉ $8$.

> Trường gợp ta giảm $\epsilon$ thì giá trị đầu ra ngày càng tiến gần đến $8$.

Reviewer: ```suggestion
Trong trường hợp ta giảm $\epsilon$ thì giá trị đầu ra ngày càng tiến gần đến $8$.
```

> Vì vậy chúng ta có thể kết luận, một cách chính xác, rằng giá trị mà chúng ta tìm kiếm (độ biến thiên đầu vào thay đổi đầu ra) là $8$ tại điểm $x=4$.

Reviewer: ```suggestion
Vì vậy chúng ta có thể kết luận một cách chính xác, rằng giá trị mà chúng ta tìm kiếm (độ biến thiên đầu vào thay đổi đầu ra) là $8$ tại điểm $x=4$.
```

> 

Author: ```suggestion
Cách mà một nhà toán học giải mã vấn đề trên là:

```

It was not solved until 1986 that the *backpropagation algorithm* introduced in :cite:`Rumelhart.Hinton.Williams.ea.1988` provided a way to calculate how *any* change of the weights together would change the loss in the same computation time as a single prediction of the network over the dataset.
-->

Một chút bàn luận ngoài lề về lịch sử: trong những thập kỷ đầu tiên của ngành nghiên cứu mạng nơ-ron, các nhà khoa học đã sử dụng thuật toán này (*phương pháp biến thiên hữu hạn*) để đánh giá một hàm mất mát dưới các nhiễu loạn nhỏ: chỉ cần thay đổi trọng số và xem cách thức mà hàm mất mát thay đổi. 

> Đây là một cách tính toán không hiệu quả, đòi hỏi hai đánh giá đối với hàm mất mát để thấy được sự tác động của một biến lên hàm mất mát đó.  

Author: Em sửa cách này đọc xuôi và thoáng nghĩa hơn hẳn. Anh thêm bớt vài từ nữa thôi:
```suggestion
Đây là một cách tính toán không hiệu quả, đòi hỏi đến hai lần tính hàm mất mát để thấy được sự tác động của một thay đổi lên hàm mất mát đó.  
```

> Nếu chúng ta sử dụng phương pháp này với vài nghìn tham số nhỏ, nó cũng sẽ đòi hỏi vài nghìn đánh giá của mạng nơ-ron trên toàn bộ dữ liệu. 

Author: ```suggestion
Nếu chúng ta sử dụng phương pháp này với vài nghìn tham số nhỏ, nó cũng sẽ đòi hỏi hàng nghìn đánh giá của mạng nơ-ron trên toàn bộ dữ liệu. 
```

> Phải đến năm 1986 thì vấn đề này với được giải quyết khi *thuật toán Lan Truyền Ngược* được giới thiệu ở :cite:`Rumelhart.Hinton.Williams.ea.1988` đã đem đến một giải pháp để tính sức ảnh hưởng của những thay đổi đồng thời *bất kỳ* từ trọng số lên hàm mất mát, mà chỉ cần mô hình sử dụng các phép tính dùng để dự đoán trên tập dữ liệu. 

Reviewer: ```suggestion
Phải đến năm 1986 thì vấn đề này với được giải quyết khi *thuật toán Lan Truyền Ngược* (*backpropagation algorithm*) được giới thiệu ở :cite:`Rumelhart.Hinton.Williams.ea.1988` đã đem đến một giải pháp để tính toán sức ảnh hưởng của những thay đổi đồng thời *bất kỳ* từ trọng số lên hàm mất mát mà chỉ cần mô hình sử dụng các phép tính dùng để dự đoán trên tập dữ liệu. 
```

> Phải đến năm 1986 thì vấn đề này với được giải quyết khi *thuật toán Lan Truyền Ngược* được giới thiệu ở :cite:`Rumelhart.Hinton.Williams.ea.1988` đã đem đến một giải pháp để tính toán sự ảnh hưởng từ các thay đổi *bất kỳ* của trọng số tới hàm mất mát trong cùng thời điểm tính toán như là dự đoán duy nhất của mạng đối với tập dữ liệu. 

Reviewer 2: ```suggestion
Phải đến năm 1986 thì vấn đề này với được giải quyết khi *thuật toán Lan Truyền Ngược* được giới thiệu ở :cite:`Rumelhart.Hinton.Williams.ea.1988` đã đem đến một giải pháp để tính sức ảnh hưởng của những thay đổi đồng thời *bất kỳ* từ trọng số lên hàm mất mát, mà chỉ cần mô hình sử dụng các phép tính dùng để dự đoán trên tập dữ liệu. 
```
Chỗ này câu gốc hơi bị dài, mình sửa lại nhưng vẫn chưa hài lòng lắm.

> Phải đến năm 1986 thì vấn đề này với được giải quyết khi *thuật toán lan truyền ngược* được giới thiệu ở :cite:`Rumelhart.Hinton.Williams.ea.1988` đã đem đến một giải pháp để tính toán sức ảnh hưởng của những thay đổi đồng thời *bất kỳ* từ trọng số lên hàm mất mát mà chỉ cần mô hình sử dụng các phép tính dùng để dự đoán trên tập dữ liệu. 

Reviewer 3: ```suggestion
Phải đến năm 1986 thì vấn đề này với được giải quyết khi *thuật toán lan truyền ngược* (_backpropagation algorithm_) được giới thiệu ở :cite:`Rumelhart.Hinton.Williams.ea.1988` đã đem đến một giải pháp để tính toán sức ảnh hưởng của những thay đổi *bất kỳ* từ các trọng số lên hàm mất mát với thời gian tính toán chỉ bằng thời gian mô hình đưa ra dự đoán trên tập dữ liệu. 
```

<!--
Back in our example, this value $8$ is different for different values of $x$, so it makes sense to define it as a function of $x$.  
More formally, this value dependent rate of change is referred to as the *derivative* which is written as
-->

> Trở lại với ví dụ của chúng ta, giá trị $8$ này biến thiên với các trị khác nhau của $x$, vậy nên sẽ là hợp lý nếu chúng ta định nghĩa nó như là một hàm của $x$. 

Reviewer: ```suggestion
Quay lại với ví dụ của chúng ta, giá trị $8$ này biến thiên với các trị khác nhau của $x$, vậy nên sẽ là hợp lý nếu chúng ta định nghĩa nó như là một hàm của $x$. 
```

For instance, all of the below notations indicate the same thing:
-->

Các văn bản khác nhau sẽ sử dụng các ký hiệu khác nhau cho đạo hàm.

> Chẳng hạn, tất cả các ký hiệu dưới đây diễn giải cùng một ý nghĩa:

Reviewer 3: ```suggestion
Chẳng hạn, tất cả các ký hiệu dưới đây đều diễn giải cùng một ý nghĩa:
```

Often times, it is intuitively useful to unravel the definition of derivative :eqref:`eq_der_def` again to see how a function changes when we make a small change of $x$:
-->

> Phần lớn các tác giả sẽ chọn một ký hiệu duy nhất và gắn bó với nó, tuy nhiên điều này cũng không phải là chắc chắn.

Reviewer 2: ```suggestion
Phần lớn các tác giả sẽ chọn một ký hiệu duy nhất sử dụng nó xuyên suốt, tuy nhiên cũng có những tác giả không như vậy.
```

> Sẽ là tốt nhất nếu chúng ta làm quen với tất cả các ký hiệu này.

Reviewer 2: ```suggestion
Tốt hơn là chúng ta nên làm quen với tất cả các ký hiệu này.
```

> Ký hiệu $\frac{df}{dx}$ sẽ được sử dụng trong toàn bộ cuốn sách này, trừ trường hợp chúng ta cần lấy đạo hàm của một biểu thức phức tạp, khi đó chúng ta sẽ sử dụng $\frac{d}{dx}f$ để biểu diễn biểu thức như sau:

Reviewer 3: ```suggestion
Ký hiệu $\frac{df}{dx}$ sẽ được sử dụng trong toàn bộ cuốn sách này, trừ trường hợp chúng ta cần lấy đạo hàm của một biểu thức phức tạp, khi đó chúng ta sẽ sử dụng $\frac{d}{dx}f$ để biểu diễn những biểu thức như
```

*

<!-- Phần 3 -->

> * Nguyễn Lê Quang Nhật

Author: ```suggestion
* Nguyễn Lê Quang Nhật
* Lê Khắc Hồng Phúc
```

*

<!-- Phần 3 -->
* Nguyễn Lê Quang Nhật
* Lê Khắc Hồng Phúc

> * Đoàn Võ Duy Thanh

Author: ```suggestion
* Đoàn Võ Duy Thanh
* Phạm Hồng Vinh
```
